tests/python/pants_test/test_base.py

```diff
@@ -616,6 +616,23 @@ def assertInFile(self, string, file_path):
       content = f.read()
       self.assertIn(string, content, '"{}" is not in the file {}:\n{}'.format(string, f.name, content))
 
+  @contextmanager
+  def assertRaisesWithMessageContaining(self, exception_type, error_text, exact=False):
```

Comment:
I think we want to default to `exact=True`. Seems better to default to the most secure/sane possible, and have to intentionally opt in to a less strong expectation.

I love that this is an option!

Reply:
I agree, will make this change (might change the name of the arg to something like `inexact` or `substring_match`).

Reply:
Changed to `substring_match=False`!